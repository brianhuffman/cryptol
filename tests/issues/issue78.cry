unique : {a,b} (a>=1) => [a][b] -> Bit
unique xs = [ exist x (i+1) | x <- xs | i <- [0..(a-1)] ] == 0
  where exist : [b] -> [width a] -> Bit
        exist x i = if(i>=`a) then False
                    else if(x==(xs@i)) then True
                    else exist x (i+1)
