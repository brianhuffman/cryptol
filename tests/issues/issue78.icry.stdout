Loading module Cryptol
Loading module Cryptol
Loading module Main
unique : {a, b} (a >= 1) => [a][b] -> Bit
